### ISSUE TITLE
vbmeta error no sign info (samsung A51 exynos 9611)

### ISSUE THREAD

**Issue by testz-99** [topjohnwu/Magisk#2272]
Hi
When using AP magisk patched tar it makes vbmeta error with dowload screen:
ODIN MODE (AVB Fail), vbmeta: error verifying vbmeta image: OK_NOT_SIGNED, VERIFICATION_DISABLED bit is set, custom vbmeta No sign info, vbmeta .
[magisk_install_log_KEEPON.log](https://github.com/topjohnwu/Magisk/files/4071184/magisk_install_log_KEEPON.log)
[magisk_install_log_KEEPOFF.log](https://github.com/topjohnwu/Magisk/files/4071185/magisk_install_log_KEEPOFF.log)

SM-A515F with A515FXXU1ASKJ firmware Android 10 has 2 vbmeta: vbmeta.img and vbmeta_samsung.img in AP.
[vbmeta_stock_imgs.zip](https://github.com/topjohnwu/Magisk/files/4071199/vbmeta_stock_imgs.zip)

Only flashing boot.img makes bootloop.
Magisk versions tested from 19.3, 20.3 to latest canary build.

TWRP is not available. The error appears after flashing and changing vbmeta.img. Bootloader is unlocked, KG is checking. With original vbmeta magisk recovery is not accepted.

What should be done to work?

**Comment by testz-99** [comment]
![vbmeta_errors](https://user-images.githubusercontent.com/59962022/72667277-f9eaef80-3a11-11ea-94fc-41bf47073850.jpg)
Still no success

**Comment by testz-99** [comment]
Some logs in [here](https://mega.nz/#F!3RJXhC7R!d-GWitMJECQmq5s24LgbMQ) to help. Only flashing boot.img from the magisk_patched.tar work but with no magisk available. Flashing only boot patched img makes bootloop. Flashing all tar makes vbmeta sign error.

**Comment by hanung665** [comment]
I'm waiting for working solution <(",)>

**Comment by raw-raw** [comment]
> I'm waiting for working solution <(",)>

Many people are. Please keep discussions to xda and away from github.com and only comment here if you have something noteworthy to add to the issue.

**Comment by kekobot** [comment]
I **faced the same problem Is there a solution?**

**Comment by ahronshor** [comment]
[here in xda](https://forum.xda-developers.com/galaxy-a51/how-to/guide-how-to-root-samsung-a51-sm-a515f-t4032389/page2) is a full tread about this issue for a51.

But the solution to make a signed recovery for each device is not so good for magisk.
Note that the same problem was also found in note 10 lite n770f,
And probably that's what will happen even in many new Samsung models...

**Comment by JPaalman** [comment]
I can confirm the issue also exists on the Note 10 Lite SM-N770F when patching the AP, but when patching the boot image only, the device is succesfully rooted and does not enter a bootloop. The downside is that the CTS profile component of SafetyNet does not pass after rooting using this method. (or at least, for me: [someone on XDA](https://forum.xda-developers.com/galaxy-note-10-lite/how-to/root-note-10-lite-magisk-fail-t4052155) claims to have rooted his phone by patching the boot image, with passing SafetyNet.)

**Comment by nehaonestore** [comment]
I have tried to patch only boot.img but its showing unable to do normal boot.

**Comment by JPaalman** [comment]
I might need to add that I did a factory restore after patching the boot image.

**Comment by osm0sis** [comment]
Is this still an issue with the current Magisk Stable?

**Comment by jhhuisman** [comment]
> Is this still an issue with the current Magisk Stable?

 I think so yes. Yesterday I spent a night trying to root the A51. Will keep an eye on this issue for updates.

**Comment by osm0sis** [comment]
You think so? I need someone to try and make sure with Magisk 20.4 stable.

**Comment by map220v** [comment]
On Stable Magisk 20.4 issue still exists 
[magisk_install_log_KEEPOFF.log](https://github.com/topjohnwu/Magisk/files/4400785/magisk_install_log_KEEPOFF.log)
![IMG_1809](https://user-images.githubusercontent.com/14368485/77875409-ad653100-7258-11ea-8a51-14fa68008664.JPG)

**Comment by osm0sis** [comment]
So just to be clear, Magisk has never worked on this device?

**Comment by map220v** [comment]
Yes. it's never worked on A51.

**Comment by nayessiva** [comment]
Hi,
I've seen that : https://github.com/topjohnwu/Magisk/issues/1961 
However I still have the issue on my A40.
I know the topic here is A51 but probably the same issue ? Or is it really fixed and i messed it up flashing magisk-patched ...?

**Comment by map220v** [comment]
I Tried to patch boot.img only on Samsung A51 (system as root) it's gives kernel panic.
On Samsung A71 (system as root) it's not gives kernel panic.
I got kmsg.txt from Samsung A51 last Magisk Canary (debug)
[kmsg.txt](https://github.com/topjohnwu/Magisk/files/4559756/kmsg.txt)

**Comment by adittya1** [comment]
I have solutions for root a515f...
![Screenshot_20200510-191634_Root Checker Basic](https://user-images.githubusercontent.com/47035354/81504599-dca89c80-92f2-11ea-9fed-eead516ebe0e.jpg)

**Comment by adittya1** [comment]
https://www.youtube.com/channel/UCBVpSYb1xfSz0LGFubdTvnA

Visite my youtube chanal, i will upload recently inshaallah this tutorial.... hurry up....

**Comment by adittya1** [comment]
https://youtu.be/aba6uH41fYg

Just demo

**Comment by adittya1** [comment]
> Yes. it's never worked on A51.

Its 2000000% work broo
![Screenshot_20200510-192011](https://user-images.githubusercontent.com/47035354/81504670-4b85f580-92f3-11ea-982e-f30d58975b28.jpg)
![Screenshot_20200510-191634_Root Checker Basic](https://user-images.githubusercontent.com/47035354/81504672-4cb72280-92f3-11ea-8cb2-0c9c66653b90.jpg)

**Comment by rodrigofd** [comment]
@osm0sis ping... this guy is spamming all around 👆🏻

**Comment by adittya1** [comment]
> @osm0sis ping... this guy is spamming all around 👆🏻

Okay....

**Comment by osm0sis** [comment]
> @osm0sis ping... this guy is spamming all around 👆🏻

So it is indeed pure bullshit?

**Comment by rodrigofd** [comment]
Anything that can be done, to help bump this issue? It's currently not possible to use Magisk on the A51.... (I could try pull my logs as well, but I don't think it would add more value to those that @map220v already posted above) 👆

**Comment by adittya1** [comment]
> > @osm0sis ping... this guy is spamming all around 👆🏻
> 
> So it is indeed pure bullshit? Removed his comments. :+1: 

Yeh no need spaming, I see that everyone need this a51 root, cs I'm say I can root. I'm come for help all, okay bro god bless you..

**Comment by map220v** [comment]
I updated phone to A515FXXU3BTD4(OneUI2.1) and now patched boot.img not gives kernel panic.
[imgur.com/AJVFInn](url)

**Comment by balanadi** [comment]
I would like to add my experience with M315F which has the same exynos 9611 and i tried to patch The entire AP-M315FXXU1ATB4 with magisk 7.5.1 and also with canary update and it gives the same error about vbmeta  , but if i patch only the boot.img it is working but stays in  bootloop  and is restarting after 10 seconds ...
Hope someone will find a way to fix this bug ... Thanks

**Comment by B1gM1k3** [comment]
Found a video how helps me installing successfully Magisk. CTS failed btw.
https://www.youtube.com/watch?time_continue=216&v=oZD2jkkKX_E&feature=emb_logo

**Comment by balanadi** [comment]
I tryed  already and it is staying in bootloop and restarting after 10 seconds ...

**Comment by B1gM1k3** [comment]
Maybe you missed a step ? I strictly followed the procedure step by step and it worked.

**Comment by balanadi** [comment]
> Maybe you missed a step ? I strictly followed the procedure step by step and it worked.

i did same as in the video , 3 times already today .. same error : restarting after factory reset and keep restarting after 10 seconds ...AP-M315FXXU1ATB4 galaxy M31F from DBT germany

**Comment by map220v** [comment]
> > Maybe you missed a step ? I strictly followed the procedure step by step and it worked.
> 
> i did same as in the video , 3 times already today .. same error : restarting after factory reset and keep restarting after 10 seconds ...AP-M315FXXU1ATB4 galaxy M31F from DBT germany

This Video won't work for you. Because you using not A51.
I recommend you to wait for Binary 2 or Binary 3 Update for your phone bootloader.

**Comment by adittya1** [comment]
I can fixed this problem.....

**Comment by adittya1** [comment]
> Found a video how helps me installing successfully Magisk. CTS failed btw.
> https://www.youtube.com/watch?time_continue=216&v=oZD2jkkKX_E&feature=emb_logo

Osm0sis  ... where are you, this is my tutorial.... do you need any help? Contact me...

**Comment by adittya1** [comment]
If anyone need root a51 if u follow tutorial, but u cannot successful, contact me, i fixed your problem i'll come u by teamviewer.... thank you.

**Comment by map220v** [comment]
> > Found a video how helps me installing successfully Magisk. CTS failed btw.
> > https://www.youtube.com/watch?time_continue=216&v=oZD2jkkKX_E&feature=emb_logo
> 
> Osm0sis ... where are you, this is my tutorial.... do you need any help? Contact me...

This video only working because you using A51 with Binary 3 update and patching boot.img instead of recovery.img

**Comment by balanadi** [comment]
Thank you guys for your help , but i will prefer to wait till a future update ...
So far i manage to build a script and uninstall all the crap bloatware without root , and without affecting knox ... so i will make a tutorial on xda when i will finish ..

**Comment by adittya1** [comment]
> Thank you guys for your help , but i will prefer to wait till a future update ...
> So far i manage to build a script and uninstall all the crap bloatware without root , and without affecting knox ... so i will make a tutorial on xda when i will finish ..

Lool

**Comment by Navneet-Kumar-Singh** [comment]
https://github.com/topjohnwu/Magisk/issues/2923#issue-646679716

**Comment by soum91** [comment]
I opened issue #2917 which was closed and I have been told by @osm0sis to add my info here, so pasting it below

> Can't bypass Secure Check Fail Screen on SM-A207F #2917
>>I am trying to flash magisk on my Samsung SM-A207F on Android 10 following this instruction. But after flashing the patched AP file I can't boot into stock recovery to Wipe data/factory reset at step 10. As soon as I press Volume Down + Power to escape Download Mode and press Volume Up + Power, it takes me to Secure Check Fail screen, no matter how many times I try. I even tried patching only Boot.img that too results in the same error. I have used Magisk 20.4. Can anyone tell how to fix this or what am I doing wrong?

I am guessing in my case it has to do something with bootloader unlock. This is a Qualcomm device and it has something called **Qualcomm Secure Boot**. Even though I have unlock OEM Bootloader, as you can see [here](https://user-images.githubusercontent.com/3830681/85872467-9ac7ad00-b7f1-11ea-93c7-f6d6d63dcf70.png), it still says **Qualcomm Secure Boot: Enabled** in bootloader. I tried googling about this but could find anything to unlock "Qualcomm Secure Boot". I am really worried about ever being able to root it!

**Comment by map220v** [comment]
@soum91 this problem has nothing to do with Qualcomm Secure Boot.
`SECURE CHECK FAIL : vbmeta` caused by vbmeta_samsung that checks vbmeta
there is only 2 fixes:

1.Patch boot.img only

2.Use original vbmeta.img and sign recovery.img using `avbtool.py add_hash_footer --image image-new.img --partition_name recovery --partition_size <original recovery.img size in bytes>`

**Comment by soum91** [comment]
@map220v I have tried flashing only boot.img so many times, always gives ** 
Secure Check Fail: boot.img**. Any kind of modifications to any of the image is triggering this.

**Comment by Navneet-Kumar-Singh** [comment]
> @map220v I have tried flashing only boot.img so many times, always gives **Secure Check Fail: boot.img**. Any kind of modifications to any of the image is triggering this.

Same here on Samsung A8 Tab Wifi Only 2019 SM-T290

**Comment by Navneet-Kumar-Singh** [comment]
> @soum91 this problem has nothing to do with Qualcomm Secure Boot.
> `SECURE CHECK FAIL : vbmeta` caused by vbmeta_samsung that checks vbmeta
> there is only 2 fixes:
> 
> 1.Patch boot.img only
> 
> 2.Use original vbmeta.img and sign recovery.img using `avbtool.py add_hash_footer --image image-new.img --partition_name recovery --partition_size <original recovery.img size in bytes>`

Can you please tell me all steps to do so ?

**Comment by mu198** [comment]
The same problem is found on the galaxy tab s6 lite (SM-P610).  It seems to be a new root lock from Samsung 🙈

**Comment by soum91** [comment]
I'm just wondering if @topjohnwu is even aware or remember about this issue. It's been more than 6 months, still haven't heard anything from John Wu.

**Comment by mu198** [comment]
Well, there were bigger problems in the world and this is still an open source project, so big thanks to @topjohnwu.  Hopefully you will tackle this problem in the near future.

**Comment by Navneet-Kumar-Singh** [comment]
> Please help me......please...............anyone...........................................................topjohnwu?
> Same issue with my Samsung Galaxy Tab A8 2019 Wifi Only (SM-T290 ) : https://github.com/topjohnwu/Magisk/issues/2923#issue-646679716 
> My device is non A/B  and is System-as-Root. 
> Tried all possible ways but the same error is coming :
> 
> The device in download mode reports: "`SECURE CHECK FAIL : vbmeta"
> 
> And at the same time Odin reports " Complete (Write) operation failed "
> Even when I used a original, non patched version of Odin same output came on my Samsung tablet.
> 
> ![part1](https://user-images.githubusercontent.com/32897976/85923001-4e2ab300-b87f-11ea-8a70-852048c4772a.png)
> ![part2](https://user-images.githubusercontent.com/32897976/85923004-5256d080-b87f-11ea-8052-62d6eeebfd63.png)
> 
> Also I noticed that Magisk patches the vbmeta.img.lz4 also along with boot.img.lz4

Update for you guys mine is binary 3 update, any suggestions advice for me? You are fully welcome.  Please help.

**Comment by Navneet-Kumar-Singh** [comment]
> Please help me......please...............anyone...........................................................topjohnwu?
> Same issue with my Samsung Galaxy Tab A8 2019 Wifi Only (SM-T290 ) : https://github.com/topjohnwu/Magisk/issues/2923#issue-646679716 
> My device is non A/B  and is System-as-Root. 
> Tried all possible ways but the same error is coming :
> 
> The device in download mode reports: "`SECURE CHECK FAIL : vbmeta"
> 
> And at the same time Odin reports " Complete (Write) operation failed "
> Even when I used a original, non patched version of Odin same output came on my Samsung tablet.
> 
> ![part1](https://user-images.githubusercontent.com/32897976/85923001-4e2ab300-b87f-11ea-8a70-852048c4772a.png)
> ![part2](https://user-images.githubusercontent.com/32897976/85923004-5256d080-b87f-11ea-8052-62d6eeebfd63.png)
> 
> Also I noticed that Magisk patches the vbmeta.img.lz4 also along with boot.img.lz4

Help me, anyone regarding this issue .

**Comment by Navneet-Kumar-Singh** [comment]
Even when I flash just boot.img.tar after patching with magisk Odin says write operation failed. As soon as the flashing is at 99 % the write operation failed. 😢😭

**Comment by Navneet-Kumar-Singh** [comment]
> Even when I flash just boot.img.tar after patching with magisk Odin says write operation failed. As soon as the flashing is at 99 % the write operation failed.

Because of this I think flashing is ok but some type of security program in bootkoader is preventing the full flash as flash WAS  about to get completed after 99 percent . What do you all say? Please reply as I am in need of solving this issue . It's too important for me and I think some others also📲

**Comment by map220v** [comment]
@Navneet-Kumar-Singh Samsung using their Odin Download Mode instead of fastboot. (Also samsung even deleted fastbootd from recovery, so flashing custom roms difficult on devices with super.img)

**Comment by Navneet-Kumar-Singh** [comment]
What is super.img? I want to know. Pls

**Comment by map220v** [comment]
@Navneet-Kumar-Singh Phones released with android 10 uses super.img. 
super.img contains system.img, vendor.img, product.img and odm.img
to install custom roms on these devices people using fastbootd recovery, but samsung removed it...

**Comment by Navneet-Kumar-Singh** [comment]
> @Navneet-Kumar-Singh Phones released with android 10 uses super.img. 
> super.img contains system.img, vendor.img, product.img and odm.img
> to install custom roms on these devices people using fastbootd recovery, but samsung removed it...

But mine is android 9 pie.

**Comment by JperiodMichael** [comment]
Magisk has been installed on a SM-T290.  There are threads on xda-developers.  In particular, @gcrutchr's instructions for SM-T295.

**Comment by soum91** [comment]
@JperiodMichael, can you please link any of the thread that's discussing about successful installation of Magisk on Android 10? I couldn't find any. I have tried every single methods I have found, nothing succeeds. Any modifications to any stock img, be it recovery, be it boot or even vbmeta won't let the phone boot, always gets looped with Secure Check Fail screen.

**Comment by TrevE** [comment]
I get the same vbmeta error / bootloop when using Magisk on the Samsung A01 on verizon running android 10 - SM-A015V_1_20200519184228_z3qv96wyic_fac .  I believe A01, A11, A21, A51, and A71 are similar.

**Comment by Navneet-Kumar-Singh** [comment]
@JperiodMichael I found this as you told but how to overlay , what is overlay , here?

Copied here from XDA:---------


I have root on my SM-T295 A8 tablet!!

Attached boot.tar file is for XXU2ASL3 only.

T295-Magisk-AP.tar

Okay, after multiple failed attempts, I read about a method to root my T295. It did not work!! But, I decided to just use Magisk Manager on the boot.img file only. The resultant file would not flash - it will fail. The original boot.img & recovery.img files are 65,536K bytes each. **I had to take the Magisk modified boot.img and overlay it onto the original # boot.ing file.** I did the same for the recovery.img. Then the files would flash successfully. The first thing after booting was to check Magisk Manager. It showed I have Magisk 20.3 installed. So, I tested this with Root Explorer. I did have root!!

If you are going to try this, you need to unlock your boot loader, I think. I have not tried this with a locked bootloader.

**Comment by Navneet-Kumar-Singh** [comment]
> @JperiodMichael I found this as you told but how to overlay , what is overlay , here?
> 
> Copied here from XDA:---------
> 
> 
> I have root on my SM-T295 A8 tablet!!
> 
> Attached boot.tar file is for XXU2ASL3 only.
> 
> T295-Magisk-AP.tar
> 
> Okay, after multiple failed attempts, I read about a method to root my T295. It did not work!! But, I decided to just use Magisk Manager on the boot.img file only. The resultant file would not flash - it will fail. The original boot.img & recovery.img files are 65,536K bytes each.**I had to take the Magisk modified boot.img and overlay it onto the original  boot.img file.** I did the same for the recovery.img. Then the files would flash successfully. The first thing after booting was to check Magisk Manager. It showed I have Magisk 20.3 installed. So, I tested this with Root Explorer. I did have root!!
> 
> If you are going to try this, you need to unlock your boot loader, I think. I have not tried this with a locked bootloader.
> 
> 

How to overlay one boot.img over other and why we do this? All suggestions welcome.

**Comment by Navneet-Kumar-Singh** [comment]
I also found this, pls help how to overlay, and how to modify kernel as mentioned in this tutorial :

By gcrutchr on XDA Forums : my Tab A8 SM-T295 with XXS3ATB2 firmware got it rooted!!Attached AP.tar file is for XXU2ASL3 only.

T295-Magisk-AP.tar

Okay, after multiple failed attempts, I read about a method to root my T295. It did not work!! But, I decided to just use Magisk Manager on the boot.img file only. The resultant file would not flash - it will fail. The original boot.img & recovery.img files are 65,536K bytes each. I had to take the Magisk modified boot.img and overlay it onto the original boot.ing file. I did the same for the recovery.img. Then the files would flash successfully. The first thing after booting was to check Magisk Manager. It showed I have Magisk 20.3 installed. So, I tested this with Root Explorer. I did have root!!

If you are going to try this, you need to unlock your boot loader, I think. I have not tried this with a locked bootloader.


Basic instructions


PLEASE READ ALL OF THIS BEFORE STARTING!! VERY IMPORTANT!!

Here are the steps to root your tablet:
Make a backup of you apps & data before doing this. This procedure will
wipe all you data.

Go to settings.
Select software info
Tap build number 7 times to get into developer mode
Go back to main settings page.
Scroll to bottom and select developer mode
In middle of page find OEM unlocking
Make sure slider is all the way to the right (should say allow bootloader to be unlocked)

Unlock bootloader:
1. turn off tablet
2. press and hold down vol up & vol down keys together
3. insert usb cable (should already be connected to computer)
4. when bootloader screen appears, release all keys
5. press and hold vol up key for 5 seconds, then release
6. read instructions on screen to unlock bootloader (this will wipe all your data)

After reboot, go through normal setup.

Now for the hard part.

You need to download the complete ROM you have installed on your Tablet.
It will be better if you have the same ROM version as I do in post in this post (#1).
You need to unzip the ROM so you can flash it in ODIN.

If you are on the same ROM version as in my post #1, then you can just download my rooted files.

Unzip my downloaded root files zip file.
Unzip the downloaded ROM files.

In the AP* file, you need to replace my files in the zip: boot.img, recovery.img & vbmeta.img

Start ODIN. Select files for BL*, AP*, CSC* & CP*
For CSC file, select HOME_CSC*
*** Uncheck auto reboot ***

Flash ROM.
After flashing ROM. press vol dn + pwr key
As soon as screen goes blank press vol up key and hold
When get into recovery, select data wipe, then reboot
Set up phone as you normally would.
After setup, go to Play store and install Magisk Manager. Run Magisk Manager to see if you have root.


Do It Yourself Instructions
You need to do some research before doing this. This is NOT a walk you by the hand tutorial. You need to learn some things before following these instructions. Learn how to use lz4.exe. Learn how to use simg2img. Learn how to make a .tar file.


Make a backup of you apps & data before doing this. This procedure will
wipe all you data.

Go to settings.
Select software info
Tap build number 7 times to get into developer mode
Go back to main settings page.
Scroll to bottom and select developer mode
In middle of page find OEM unlocking
Make sure slider is all the way to the right (should say allow bootloader to be unlocked)

Unlock bootloader:
1. turn off tablet
2. press and hold down vol up & vol down keys together
3. insert usb cable (should already be connected to computer)
4. when bootloader screen appears, release all keys
5. press and hold vol up key for 5 seconds, then release
6. read instructions on screen to unlock bootloader (this will wipe all your data)

After reboot, go through normal setup.

Now for the hard part.

You need to download the complete ROM you have installed on your Tablet.
Extract ROM files
Extract system.*.lz4, recovery*.lz4 and vbmeta*.lz4
Use tar to make an AP.tar file with the 3 above files. **Search the Internet if you do not know how to use tar.
Copy the AP.tar file to your phone. It is better to use adb push to prevent file correuption.
Install Magisk Manager from Play store.
Run Magisk Manager. Select install Magisk. Choose Select and Patch a file. Follow directions on screen to patch the AP.tar file.
Once the AP.tar file has been patched, use adb to pul the file to yur computer.
Extract the file from the AP.tar file.

Now come the hard part:

You cannot edit .lz4 directly. You have to decompress them. After decompressing system*, use simg2img to decompress the sparse file. Search the Internet if you do not know how to do this.

Now we make our modified files so they will flash in ODIN.

Instructions:
I used hex editor neo - a windows 10 app
**I opened the stock kernel & recovery files in hex editor neo
I opened the Magisk modified kernel & recovery files in hex editor neo
I copied the modified kernel & recovery code onto their respective stock files in hex editor neo
I saved the stock files & closed hex editor neo.**
I used cygwin to create a tar file: boot.img, recovery.img, vbmeta.img plus other files in original AP*.MD5 file.
You should also flash all ROM *.MD5 files (BL, CP, CSC) per Magisk instructions.
New AP.tar contents:
boot.img
dtbo.img.lz4
meta-data
recovery.img
system.img.ext4.lz4
userdata.img.ext4.lz4
vbmeta.img
vendor.img.ext4.lz4

I used Odin to flash the tar file

Flash ROM.
After flashing ROM. press vol dn + pwr key
As soon as screen goes blank press vol up key and hold
When get into recovery, select data wipe, then reboot
Set up phone as you normally would.
After setup, go to Play store and install Magisk Manager. Run Magisk Manager to see if you have root.

After installing Magisk:

(Powering up normally) → (System with NO Magisk)
(OEM Recovery Key Combo) → (Splash screen) → (Release all buttons) → (System with Magisk)
(OEM Recovery Key Combo) → (Splash screen) → (Keep pressing volume up) → (Actual recovery)

Please follow my instructions and the install instructions from Magisk web page.

Magisk Installation

**Comment by JperiodMichael** [comment]
I'm sorry, this was the wrong place for me to post.  I saw #2923.  It was closed with the comment that it was a subset of this issue.

I'm not aware of any success installing Magisk on a SM-T290 running Android 10.  My tablet is running Android 9 (Pie).

@Navneet-Kumar-Singh:  I think it would be more appropriate for you to post on xda-developers.  There's a thread  [Root for SM-T290 (tab a 8.0 2019)](https://forum.xda-developers.com/galaxy-tab-a/how-to/root-sm-t290-tab-8-0-2019-t3965624).  Details of @gcrutchr's method are in his thread [SM-T295 - Finally rooted tablet!! ](https://forum.xda-developers.com/galaxy-tab-a/how-to/sm-root-t4050329).

**overlay** was:  make a copy of the stock boot or recovery image; overwrite the beginning of that file with the Magisk patched image.  The thinking was that, since the Magisk patched image was substantially smaller than the original image, there must have been something important in the "remainder" of the original image.

**Comment by gcrutchr** [comment]
> I'm sorry, this was the wrong place for me to post. I saw #2923. It was closed with the comment that it was a subset of this issue.
> 
> I'm not aware of any success installing Magisk on a SM-T290 running Android 10. My tablet is running Android 9 (Pie).
> 
> @Navneet-Kumar-Singh: I think it would be more appropriate for you to post on xda-developers. There's a thread [Root for SM-T290 (tab a 8.0 2019)](https://forum.xda-developers.com/galaxy-tab-a/how-to/root-sm-t290-tab-8-0-2019-t3965624). Details of @gcrutchr's method are in his thread [SM-T295 - Finally rooted tablet!! ](https://forum.xda-developers.com/galaxy-tab-a/how-to/sm-root-t4050329).
> 
> **overlay** was: make a copy of the stock boot or recovery image; overwrite the beginning of that file with the Magisk patched image. The thinking was that, since the Magisk patched image was substantially smaller than the original image, there must have been something important in the "remainder" of the original image.

Yes, there is information at the end of the boot.img & the recovery.img files. It is needed to install the images.

**Comment by Navneet-Kumar-Singh** [comment]
How to overlay patchedboot.img onto stockboot.img pls tell me with steps ? I want to know .

**Comment by gcrutchr** [comment]
> How to overlay patchedboot.img onto stockboot.img pls tell me with steps ? I want to know .

Read Post #1 of my thread. It is explained there. Cannot be more simple. I do not make videos.

**Comment by Navneet-Kumar-Singh** [comment]
> > How to overlay patchedboot.img onto stockboot.img pls tell me with steps ? I want to know .
> 
> Read Post #1 of my thread. It is explained there. Cannot be more simple. I do not make videos.

At least tell me the lines of code to write for overlaying.

**Comment by soum91** [comment]
>overwrite the beginning of that file

@gcrutchr what beginning are you talking about? HEX code? There are no different parts, only single continuous code of 11669312 lines. How much of that whole code one should replace?

**Comment by gcrutchr** [comment]
> > overwrite the beginning of that file
> 
> @gcrutchr what beginning are you talking about? HEX code? There are no different parts, only single continuous code of 11669312 lines. How much of that whole code one should replace?

See attached video....so simple!!
[https://drive.google.com/file/d/1yBNAA-o5Mro3hOLZlwjA3NVD27A7_4TG/view?usp=sharing](url)

**Comment by soum91** [comment]
@gcrutchr, sadly your method doesn't work too! Google/Samsung has fucked us so hard this time. Apparently NOTHING IN EXISTENCE WORK for Samsung device on Android 10. I don't anything changes soon until @topjohnwu himself fixes this! I have been trying for last 1 week but failed countlessly ! I'm done with this shit! If John Wu ever pushes any fix, someone please ping me.

**Comment by gcrutchr** [comment]
> @gcrutchr, sadly your method doesn't work too! Google/Samsung has fucked us so hard this time. Apparently NOTHING IN EXISTENCE WORK for Samsung device on Android 10. I don't anything changes soon until @topjohnwu himself fixes this! I have been trying for last 1 week but failed countlessly ! I'm done with this shit! If John Wu ever pushes any fix, someone please ping me.

All you had to do is ask me about Android 10. I already tried rooting. It didn't work. Samsung changed something. Am still investigating issue.

**Comment by mu198** [comment]
Samsung simply added an in-house version of the vbmeta that only works with the pure vbmeta, as long as magisk cannot patch them, it will try root.  Always give problems on samsung devices.  Yes, a twrp would be nice

**Comment by mu198** [comment]
Or am I so wrong now?

**Comment by Navneet-Kumar-Singh** [comment]
> > > overwrite the beginning of that file
> > 
> > @gcrutchr what beginning are you talking about? HEX code? There are no different parts, only single continuous code of 11669312 lines. How much of that whole code one should replace?
> 
> See attached video....so simple!!
> [https://drive.google.com/file/d/1yBNAA-o5Mro3hOLZlwjA3NVD27A7_4TG/view?usp=sharing](url)
> 

Thank you for making a video

**Comment by mu198** [comment]
I have now managed to root my tablet (SM-P610 / S6 Lite) 
1) download firmware 
2) extract the AP file 
3) extract Boot.lz4 ( https://github.com/mcmilk/7-Zip-zstd/Releases ) 
4) extract it to Boot.img 
5) Flash with magisk 
6) magisk_patched.tar rename to Boot.img 
7) package the Boot.img into a .tar file (via 7z) 
8) Flash Boot.tar via Odin 3.14 
8) then immediately perform the recovery and factory reset


Video https://youtu.be/SKYJL9BqZMY

Is a German tutorial

**Comment by soum91** [comment]
@mu198 what Android version your tab is on? Is it on Android 10? I can't remember but I think I have tried this method too. If you are not on Android 10, I don't think this method won't work on 10.

**Comment by mu198** [comment]
It is android 10 06/2020 Patch

**Comment by mu198** [comment]
@soum91

**Comment by soum91** [comment]
**NOPE! SAME SHIT AGAIN!** 

 I will not blame you, @mu198 . Your device is most probably not SAR device. Otherwise it’s not possible to root using said method. It’s my fault I wasted my time and setup going through same shitshow! Nothing will work until @topjohnwu finds a way out to this or TWRP team releases a recovery for SAR/Samsung device on android 10. **TILL THEN THERE WILL NOT BE ANY OPTION TO ROOT THESE DEVICES! So everyone stop wasting your time!**

![Photo on 16-7-20 at 2 57 AM](https://user-images.githubusercontent.com/3830681/87597392-8a7d6080-c713-11ea-9482-5137712c4787.jpg)

**Comment by mu198** [comment]
@soum91 

oh okay, i'm sorry, my system is a SAR.  have attached a screenshot of the latest canary manager, which is unfortunately very buggy with me, hence the stable 7.5.1 version

![20200716_050928.jpg](https://user-images.githubusercontent.com/65677888/87622062-8efd4580-c722-11ea-99a8-af06e1dd7b91.jpg)

**Comment by JperiodMichael** [comment]
I have now seen claims of installing Magisk on Android 10, on the SM-T295:
https://forum.xda-developers.com/showpost.php?p=82995753&postcount=95

He says he forcibly downgrades the bootloader while running Android 9, before installing Android 10.  Then he installs Magisk into the boot partition.  I don't understand how he gets away with it.  Does the SM-T295 *stop* being system-as-root in Android 10?

**Comment by soum91** [comment]
> I have now seen claims of installing Magisk on Android 10, on the SM-T295:
> https://forum.xda-developers.com/showpost.php?p=82995753&postcount=95
> 
> He says he forcibly downgrades the bootloader while running Android 9, before installing Android 10. Then he installs Magisk into the boot partition. I don't understand how he gets away with it. Does the SM-T295 _stop_ being system-as-root in Android 10?

@gcrutchr can you explain the process please? Did you flash BL file of Android 9 or something? What do you exactly mean by using previous bootloader? Another video like last time will be helpful but not really necessary. A bit explanation will do.

**Comment by soum91** [comment]
> I have now seen claims of installing Magisk on Android 10, on the SM-T295:
> https://forum.xda-developers.com/showpost.php?p=82995753&postcount=95
> 
> He says he forcibly downgrades the bootloader while running Android 9, before installing Android 10. Then he installs Magisk into the boot partition. I don't understand how he gets away with it. Does the SM-T295 _stop_ being system-as-root in Android 10?

@gcrutchr can you explain the process please? Did you flash BL file of Android 9 or something? What do you exactly mean by using previous bootloader? Another video like last time will be helpful but not really necessary. A bit explanation will do.

**Comment by gcrutchr** [comment]
After installing Android 10, install the BL from previous Android 9.
Use Magisk Manager to modify: Boot, recovery, vbmeta
Create .tar file from boot & vbmeta only. Flash .tar file in ODIN. Voila!! You are now rooted.

**Comment by soum91** [comment]
> Create .tar file from boot & vbmeta only. Flash .tar file in ODIN. Voila!! You are now rooted.

@gcrutchr are you sure your device is SAR? Because John Wu himself says:

> 
> Magisk actually patches 3 partitions on your device:
> 
>     vbmeta: replace with empty vbmeta image to disable partition verification
>     boot: remove the signature of the image to prevent soft bricks
>     recovery: this is where Magisk is actually installed

So if you flash boot & vbmeta, it won't give you root at least through Magisk! Because for SAR device, you need to install magisk into recovery. Same goes with @mu198's method. Patching a boot image alone won't/shouldn't give you root on SAR device on android 10 because here magisk is installed in recovery partition as per John Wu.

Still I gave a try. Flashed bootloader of Android 9. At first I tried using your method, it still didn't boot because same old reason of it being modified boot. I even tried flashing whole magisk patched AP file with android 9 bootloader and that gave the same result! I also gave a try flashing a tar created with Vbmeta file from android 9, that gave bootloop unfortunately!

If possible, a video of the whole process like last time will be really helpful as things have failed even after following your instruction. 
 
![image](https://user-images.githubusercontent.com/3830681/87775798-7f781c80-c848-11ea-80ec-5ec5746031bb.png)

**Comment by map220v** [comment]
On Samsung A51(SAR) starting from Binary 3 Update we just patching boot.img and Magisk works.

Also it's possible to install patched recovery.img but it's only works with original vbmeta and recovery.img with avb sign
i used this command to sign patched recovery.img `avbtool.py add_hash_footer --image magisk_recovery.img --partition_name recovery --partition_size <original recovery.img size in bytes>`

**Comment by mu198** [comment]
@soum91

I did not root magisk in the recovery![Screenshot_20200717-130203.jpg](https://user-images.githubusercontent.com/65677888/87779991-0e2b7000-c82e-11ea-9e65-8d46c66f3a57.jpg), magisk automatically selected these two for me and then patched it .... Because magisk says of its own accord what he wants to patch into the Boot.img or .tar.

**Comment by soum91** [comment]
> i used this command to sign patched recovery.img `avbtool.py add_hash_footer --image magisk_recovery.img --partition_name recovery --partition_size <original recovery.img size in bytes>`

@map220v can you please elaborate how to sign the img on Windows? Also do you have the avbtool.py file?

**Comment by map220v** [comment]
@soum91 https://android.googlesource.com/platform/external/avb/+/master/avbtool.py
it's works only on linux or WSL

**Comment by soum91** [comment]
@map220v, I see! That's why it's giving me syntax error both on macOS and Windows. I don't if it's appropriate for me to ask but if I provide two img can you please sign it for me on your system?

**Comment by map220v** [comment]
@soum91 I don't have access to linux right now. 
But i think you can install python in termux and sign image on phone.

**Comment by soum91** [comment]
@map220v no issue. But can you clear one thing? Do I need to enter the size of recovery.img.lz4  file or I need to extract the lz4 and then overwrite the size of the extracted img? Because if I use the size of lz4 file avbtool says size is not multiple of block size 4096.

Also Avbtool gives error if you use python3, if anyone is wondering. That's it was giving me syntax error. Otherwise Avbtool should work on macOS and windows too. Though I have not tried with python2 on them, so can't confirm.

**Comment by map220v** [comment]
@soum91 use size of recovery.img (extracted recovery.img.lz4) for me it was 71106560 bytes

**Comment by soum91** [comment]
@map220v, Yes I did that. Well, this time recovery flashed without any problem but Magisk Manager still shows magisk not installed.
You were saying you installed magisk just patching boot img. Did that involve avbtool singing of boot img too? Can you say what you did there?

**Comment by map220v** [comment]
> @map220v, Yes I did that. Well, this time recovery flashed without any problem but Magisk Manager still shows magisk not installed.
> You were saying you installed magisk just patching boot img. Did that involve avbtool singing of boot img too? Can you say what you did there?

Try reboot to recovery option in magisk or `adb reboot recovery` maybe it can boot magisk.

I just patched boot.img with magisk and flashed.
But if it's not working you can try to sign boot.img

**Comment by osm0sis** [comment]
If I recall correctly avbtool does work with Python2

**Comment by soum91** [comment]
@osm0sis you are right. It only worked when I used python2(on termux, didn't try on macOS/windows). BTW, have got any word from John Wu on this issue? Or is anyone working or will work on this issue?

**Comment by osm0sis** [comment]
Well it's still an open issue, isn't it? 😉

**Comment by soum91** [comment]
Well, I'm not complaining but the issue was opened in January and has been opened for for 7 months. A hint of it still being worked would give us frustrated soul a bit peace. Sorry if I sounded entitled. I fully appreciate the effort you guys put in this project. ☺️

**Comment by TheK0tYaRa** [comment]
> On Samsung A51(SAR) starting from Binary 3 Update we just patching boot.img and Magisk works.
> 
> Also it's possible to install patched recovery.img but it's only works with original vbmeta and recovery.img with avb sign
> i used this command to sign patched recovery.img `avbtool.py add_hash_footer --image magisk_recovery.img --partition_name recovery --partition_size <original recovery.img size in bytes>`

In the process of flashing got `Cause: init:1 empty caller`
Otherwise the process was the same, the phone returned vbmeta error `AVB fail`

**Comment by TheK0tYaRa** [comment]
Used stock vbmeta.img instead of patched one, it booted.

**Comment by TheK0tYaRa** [comment]
No trace of magisk somehow. Recovery boot gets me into recovery.

**Comment by TheK0tYaRa** [comment]
The phone doesn't wish to turn off from recovery while it does so from download mode. Fun but useless.

**Comment by TheK0tYaRa** [comment]
Apparently the "install to recovery" tick doesn't work because it doesn't change the resulting patched tar

**Comment by Janer1** [comment]
I was finally able to flash a "blank" vbmeta but it must be created via AVB tool and you need to untick Perserve AVB 2,0/dm-verity and file encryption while patching the boot.img. Samsung also changed the fstab to the Vendor parition instead so once rooted you can manually remove  the word "avb" from the fstab. I only did this on the s6 tab lite by the way. took me a week but finally got a blank vbmeta to work :D

**Comment by soum91** [comment]
@Janer1, can you please elaborate the process? How did you use Avbtool to create empty vbmeta? What command did you use? What magisk version did you use? Is you device's processore? Exynos or Snapdragon?

**Comment by Janer1** [comment]
It is a exynos devices with the same processor the S6 Tab Lite. Im guessing the avbtool created a new version of vbmeta as the old version is not compatible with it.  Again i basically patched the boot.img by extracting the boot.img via lz4 and then patching it with magisk with Perserve AVB2.0/DM-verity off and force encryption off by unticking them. Then i flashed the boot.img and it booted, it takes a while to boot at first so it may seem like a bootloop unless it keeps restarting it self. You  can try flashing this Vbmeta first then flashing a patched boot.img for the device.
[vbmeta.zip](https://github.com/topjohnwu/Magisk/files/4960018/vbmeta.zip)
Just extract the tar file and flash it via AP

**Comment by soum91** [comment]
Thanks, @Janer1. I don't think vbmeta specific to device but my device is Snapdragon one. But if that matters, can you tell me what command did you use to create empty vbmeta? Or the steps? I know it worked for you but empty vbmeta created by magisk manager also triggers the system. I'm really tired of all those 100s failes attempts.

**Comment by mu198** [comment]
@soum91 
 Do you have an unlock bootloader, snapdragon is only in the US right?  And verizon devices cannot unlock the bootloader.

**Comment by soum91** [comment]
@mu198, yes my bootloader is unlocked.

**Comment by soum91** [comment]
@Janer1, nope! It says 
```
Secure check fail 
invalid percent of drawing download progress
invalid percent of drawing download progress
Phone doesn't even boot!

**Comment by Janer1** [comment]
> @Janer1, nope! I didn't work. It says
> `Secure check fail`
> `invalid percent of drawing download progress`
> `invalid percent of drawing download progress `
> 
> It doesn't even boot to download/fastboot mode. Straight bootloop. I think I have to try by creating empty vbmeta, if you tell me the process. Still doubt that will work because that is very root cause for which this issue has been opened in the first place.

Oh I don’t think you can root snapdragon devices

**Comment by mu198** [comment]
Jupp, on the snapdragon devices, many bite their teeth, is not impossible but very, very difficult

**Comment by soum91** [comment]
> > @Janer1, nope! I didn't work. It says
> > `Secure check fail`
> > `invalid percent of drawing download progress`
> > `invalid percent of drawing download progress `
> > It doesn't even boot to download/fastboot mode. Straight bootloop. I think I have to try by creating empty vbmeta, if you tell me the process. Still doubt that will work because that is very root cause for which this issue has been opened in the first place.
> 
> Oh I don’t think you can root snapdragon devices. If you can it’ll be very hard. I’ve never seen an unlocked Samsung device on snapdragon


> Jupp, on the snapdragon devices, many bite their teeth, is not impossible but very, very difficult

SERIOUSLY? Isn't there any samsung snapdragon device that has been rooted? Won't it ever be possible? Why has no one told me before i tried for thousandth time?
Why the fk have I bought this crap! FML!

**Comment by gcrutchr** [comment]
> Jupp, on the snapdragon devices, many bite their teeth, is not impossible but very, very difficult

Snapdragon phones are very easy to root!! I have rooted my note 10 and note 10+. Am on qualcomm SD855 snapdragon with android 10

**Comment by mu198** [comment]
And how did you root it?

**Comment by soum91** [comment]
@gcrutchr, apparently what Janer1, mu198 are telling is true word by word. At least for late 2019 device. SM-A207F device has been released for almost 1year, no one has been able to root this device yet. I have been trying for like 1month, tried everything suggested by everyone. I tried everything suggested by you, from hex editing to using Android 9 BL, nothing has worked. Sadly this device is not even listed in xda, let alone a forum. I don't think this device will rooted anytime soon. Wish I could build twrp for this device. But that's long process, I don't have single ounce of patience for anything!

**Comment by mu198** [comment]
@soum91 Sry 😞

**Comment by gcrutchr** [comment]
> And how did you root it?

Install twrp, install Magisk 20.4. reboot phone

**Comment by mu198** [comment]
Yes okay, if there is a twrp it is easy

**Comment by TheK0tYaRa** [comment]
> Straight bootloop. 

Do you mean that it's not booting to recovery or is it safe to check?

**Comment by TheK0tYaRa** [comment]
So far result is the phone ASKED for factory reset.

**Comment by TheK0tYaRa** [comment]
Maybe i should flash other files along with modified vbmeta and patched boot. Currently soft bootloop.
At least it has no "vbmeta error"

**Comment by TheK0tYaRa** [comment]
Adding both patched boot and modified vbmeta to one tar with other files from AP and flashing it with others, then doing factory reset was a bad idea. Currently going to recovery is impossible. send help.

**Comment by soum91** [comment]
@TheK0tYaRa bootloop means the phone didn't complete boot, stayed forever at Samsung logo. It didn't even show the Safety check error screen. Didn't boot to download mode, only booted in fastboot mode. What is your device? Is it a snapdragon one? When you flashed the empty vbmeta didn't it give

@osm0sis I think I have found something. I don't think magisk is patching the img correctly. Because I tried to check image info using abvtool, it said it was not a correct vbmeta file. What I have noticed if an img file doesn't have partition specified in the property, even though it shows flashing has completed normally it in reality skips flashing the img. As a result magisk never gets installed but you get the impression that img was flashed. But when you sign the img using abvtool and specify the partition, it flashes the img properly and avb gets triggered hence shows boot lock up screen.

**Comment by TheK0tYaRa** [comment]
This bootloop is the one when it goes to red message and after 5 seconds restarts. I am able to flash original firmware back. I think.

**Comment by soum91** [comment]
@TheK0tYaRa, you have to go to fastboot mode and flash all 4 stock file. For that press all vol+,-power button together. It should take you to the BL unlock screen from there select recovery. Also try not to flash AP alone ever. It shrinks the data partition. But doing a factory reset from recovery should fix that.

**Comment by mu198** [comment]
Did you TAKE the canary image?  Does your magisk have to be installed in your recovery?  What does magisk say?

**Comment by TheK0tYaRa** [comment]
> Did you TAKE the canary image? Does your magisk have to be installed in your recovery? What does magisk say?

1 - yes,2 - don't know,3 - it patched fine

**Comment by TheK0tYaRa** [comment]
The next thing i'm gonna try is patched recovery and boot, modified vbmeta and all the other files as is.

**Comment by soum91** [comment]
@TheK0tYaRa best of luck with that. I have done that long before. Didn't do anything. Hope it works for you.

**Comment by mu198** [comment]
Is it really a bootloop with Boot.img?  My tray was 10 minutes in boot mode, after the factory reset, it could take a while

**Comment by TheK0tYaRa** [comment]
If it reboots ten times in a row - it's bootloop, ok?

**Comment by TheK0tYaRa** [comment]
After replacing patched `vbmeta.img` with `vbmeta.tar` from @Janer1 and flashing all 4 files of firmware with modified AP i got a Brand New error!

```
recovery: Error validating foot____(0)
recovery: Error verifying vbmeta image: invalid vbmeta header (6)
CUSTOM RECOVERY
VBMETA M215FXXU1ATF4, 31959864R
```

**Comment by soum91** [comment]
@TheK0tYaRa did you patched boot, recovery separately? Did you turned recovery mode on ever? Did you keep Preserve AVB 2.0/dm-verity, Preserve force encryption turned on? Did you replace boot, recovery, vbmeta of AP with the patched one or created new tar to flash?

**Comment by Janer1** [comment]
> After replacing patched `vbmeta.img` with `vbmeta.tar` from @Janer1 and flashing all 4 files of firmware with modified AP i got a Brand New error!
> 
> ```
> recovery: Error validating foot____(0)
> recovery: Error verifying vbmeta image: invalid vbmeta header (6)
> CUSTOM RECOVERY
> VBMETA M215FXXU1ATF4, 31959864R
> ```
> 
> where _____ is front camera



Did you just try the boot img and vbmeta only?  You can try building it your self using the Avbtool you need Linux though and python installed . Also I never patched recovery as I believe you can’t.

**Comment by Janer1** [comment]
> @TheK0tYaRa did you patch boot, recovery separately? Did you turn recovery mode on ever? Did you keep Preserve AVB 2.0/dm-verity, Preserve force encryption turned on? Did you replace boot, recovery, vbmeta of AP with the patched one or created new tar to flash?

I had those turned off and it worked.

**Comment by Janer1** [comment]
I only patched boot.img and kept Preserve Avb and force encryption off

**Comment by soum91** [comment]
@Janer1 what command do use to create empty vbmeta? I have asked you quite a few times, didn't get answer. It will great help if tell that.

**Comment by Janer1** [comment]
> @Janer1 what command do use to create empty vbmeta? I have asked you quite a few times, didn't get answer. It will great help if tell that.

avbtool make_vbmeta_image --out vbmeta.img

**Comment by TheK0tYaRa** [comment]
@soum91 Patch recovery checked, others unchecked, patched full AP tar, replaced vbmeta with the one specified above.
@Janer1 I replaced vbmeta myself and everything else before that is magisk's work. I live on Manjaro so that's no problem.

**Comment by soum91** [comment]
@Janer1 so as I said, vbmeta has nothing to do with device config but still it gave me `invalid percent of drawing download progress `. It definitely has something to do with snapdragon or newer device.

**Comment by Janer1** [comment]
> @Janer1 so as I said, vbmeta has nothing to do with device config but still it gave me `invalid percent of drawing download progress `. It definitely has something to do with snapdragon or newer device.

Yeah snapdragon devices are hard to crack after they have it’s easier

**Comment by mu198** [comment]
But strangely only with Samsung models

**Comment by TheK0tYaRa** [comment]
So, to be clear - should i sing boot and recovery with avbtool before adding them to AP.tar? Should i zero-sign vbmeta?

**Comment by Janer1** [comment]
> So, to be clear - should i sing boot and recovery with avbtool before adding them to AP.tar? Should i zero-sign vbmeta?

You could try I only patched boot.img for my tablet but I think the a51 is different.  You can try signing both if not just boot. Img and flash only that.

**Comment by TheK0tYaRa** [comment]
zero-signing them or signing them with vbmeta size?

**Comment by Janer1** [comment]
> zero-signing them or signing them with vbmeta size?

You could try both

**Comment by TheK0tYaRa** [comment]
Basically the root of the problem is vbmeta file. It needs to be both signed and empty. On that point the extent of my knowledge ends. Gonna try that asap.

**Comment by TheK0tYaRa** [comment]
1 hour in after odin started flashing, stuck on prism. Send help.
[vbmeta.zip](https://github.com/topjohnwu/Magisk/files/4974737/vbmeta.zip)

**Comment by TheK0tYaRa** [comment]
used 1mb one

**Comment by TheK0tYaRa** [comment]
16kb flashed fine, `Public key used to sign____a rejected (5)` is a change that i'm willing to take because i've forgot to sign recovery.

**Comment by TheK0tYaRa** [comment]
Welp, it wasn't recovery - that's for sure.

**Comment by soum91** [comment]
@TheK0tYaRa, any update?

**Comment by TheK0tYaRa** [comment]
Haven't tried anything since trying to flash TWRP fork OrangeFox. It bootlooped to "emergency recovery" mode. If there is no way to flash or sideload it properly then RIP

**Comment by TheK0tYaRa** [comment]
But the recovery was built by me so it might be just broken.

**Comment by TheK0tYaRa** [comment]
[Here](https://drive.google.com/file/d/1o4Xp5261pJId_FvI4keT-VzNGeA7DKHw/view?usp=sharing) and [here](https://drive.google.com/file/d/1oAoiXm-nbYDFb_wgy_GdsIzz0niU9tjw/view?usp=sharing) are built recoveries, if someone on samsung galaxy M21 aka REL wishes to check why it bootloops - you're welcome to do so, i don't know anything about why it is so. Sideloading the zip results in `missing footer` error 21 so it's safe. If you wish to try sign it - go for it!

**Comment by 4ifi** [comment]
Patched boot.img on SM-M215F and got bootloop.
https://yadi.sk/d/LzF8Z4UPt_828w

And also tried to patch AP and got download mode loop

**Comment by mu198** [comment]
Did you rename the magisk_patched.tar to Boot.tar when you flashed it?

**Comment by TheK0tYaRa** [comment]
> Patched boot.img on SM-M215F and got bootloop.
> And also tried to patch AP and got download mode loop

It's currently a normal behavior. Those three vbmeta.img files do quite a beating.

**Comment by TheK0tYaRa** [comment]
New info! Patching both vbmeta.img in AP and BL and also patching vbmeta_samsung.img in AP results in error while flashing. The error is about the last img having no sign info (obliviously). But the error occurs only when ending to flash PRISM.

**Comment by soum91** [comment]
@TheK0tYaRa, screw it. This is not going to be possible through Magisk. It hasn't happened in 8 months, won't happen anytime soon if ever. 
If you can just build twrp for your device. Wish I could do that. But problem is it needs you to have linux and 50GB+ storage which is viable for my virtual linux on my MacBook pro.
If you have the resources, go follow this [guide](https://github.com/SebaUbuntu/TWRP-device-tree-generator/wiki) and build you TWRP.

**Comment by map220v** [comment]
> I'm gonna have a heart attack. What does ro.arch stand for? I don't know. Why is it exynos9610 ? I don't know again. Is Samsung screwing their buyers or something?

Samsung didn't made Firmware for 9611 from scratch they used 9610 as base.

**Comment by map220v** [comment]
Magisk started working on A51 only after Binary 3 update (boot.img patch). Before Binary 3 A51 had kernel panic after patching only boot.img

**Comment by TheK0tYaRa** [comment]
I am going to assume heimdall is of no use because after flashing stock firmware with it a bootloop occured. And because it haven't been updated for two years.

**Comment by map220v** [comment]
@TheK0tYaRa heimdall can't flash images bigger than 4gb. super.img around 5gb

**Comment by TheK0tYaRa** [comment]
Then i wonder why it succeeded.

**Comment by TheK0tYaRa** [comment]
> build you TWRP.

Built OrangeFox (based on TWRP). Had issues flashing it. Replacing recovery.img failed via bootloop.

**Comment by soum91** [comment]
Isn't OrangeFox a fork? The guide link I posted is for original TWRP. You should give this one a try.

**Comment by TheK0tYaRa** [comment]
> Isn't OrangeFox a fork? The guide link I posted is for original TWRP. You should give this one a try.

I did so. It gave an error unlike OrangeFox and i didn't find the solution to it yet.

**Comment by TheK0tYaRa** [comment]
So currently i've stumbled onto some info about that the vbmeta_samsung.img should be unaltered but the images should be signed by it's stored signs.

**Comment by TheK0tYaRa** [comment]
So i've almost succeeded at building TWRP but [this](https://del.dog/ivustanuhi) happened. Why exactly do adb and fastboot offend building process - i don't know.

**Comment by mu198** [comment]
I found it in a Telegram group, maybe it will help you

https://t.me/twrp_Mauronofrio/39383

**Comment by soum91** [comment]
That group & the dev doesn't deal with shitty Samsung devices unfortunately!

**Comment by mu198** [comment]
Yes, I know the group from my oneplus 7 and I found the twrp to test or have a look, it will be fine 🙈

**Comment by JperiodMichael** [comment]
> 
> 
> So i've almost succeeded at building TWRP but [this](https://del.dog/ivustanuhi) happened. Why exactly do adb and fastboot offend building process - i don't know.
> UPD1 oh, embedded.mk

Did you move adb and fastboot to PRODUCT_HOST_PACKAGES from PRODUCT_PACKAGES?

**Comment by TheK0tYaRa** [comment]
The current [error](https://del.dog/ytottinova) is at the `mka recoveryimage` stage, i was not able to move further that error.

**Comment by Janer1** [comment]
> The current [error](https://del.dog/ytottinova) is at the `mka recoveryimage` stage, i was not able to move further that error.

Did you try creating the exports folder?

**Comment by JperiodMichael** [comment]
> 
> 
> > The current [error](https://del.dog/ytottinova) is at the `mka recoveryimage` stage, i was not able to move further that error.
> 
> Did you try creating the exports folder?

I think he needs to define a macro to indicate where he put the downloaded library.

**Comment by TheK0tYaRa** [comment]
Software update has come, it's now PDA:TG3 / CSC:TG1 / PHONE:TG1.

> Did you try creating the exports folder?

I will try to do that as fast as i'll understand what that exactly means.

> I think he needs to define a macro to indicate where he put the downloaded library.

If you mean android sdk  - it's in the binaries list.

**Comment by JperiodMichael** [comment]
> 
> 
> Software update has come, it's now PDA:TG3 / CSC:TG1 / PHONE:TG1.
> 
> > Did you try creating the exports folder?
> 
> I will try to do that as fast as i'll understand what that exactly means.
> 
> > I think he needs to define a macro to indicate where he put the downloaded library.
> 
> If you mean android sdk - it's in the binaries list.

FAILED: ninja: 
'/home/thek0tyara/git/platform_manifest_twrp_omni/10.0/out/target/product/REL/obj/SHARED_LIBRARIES/libinit_intermediates/export_includes', needed by 
'/home/thek0tyara/git/platform_manifest_twrp_omni/10.0/out/target/product/REL/obj/EXECUTABLES/recovery_intermediates/import_includes',
 missing and no known rule to make it

     Is "export_includes" something you were supposed to download?  Maybe another git repository.  Maybe you didn't put it in quite the right place in your source tree.  How much of the paths of the missing piece do you have in your source tree?  (You clearly have "/home/thek0tyara/git/platform_manifest_twrp_omni/10.0/out/target/product/REL/obj/".  Is there a "SHARED_LIBRARIES" under that?  A "libinit_intermediates" under that?)

Maybe you are supposed to build "export_includes" before trying to build whatever it is that gave you the error message.

**Comment by TheK0tYaRa** [comment]
> You clearly have "/home/thek0tyara/git/platform_manifest_twrp_omni/10.0/out/target/product/REL/obj/". Is there a "SHARED_LIBRARIES" under that? A "libinit_intermediates" under that?
> 
> Maybe you are supposed to build "export_includes" before trying to build whatever it is that gave you the error message.

The device tree was generated via [SebaUbuntu's tool](https://github.com/SebaUbuntu/TWRP-device-tree-generator) and corrected to semi-work. There is only `CONFIG` folder in the `obj` folder, there is `kati_packaging` and there is `disk.mk` .

**Comment by Janer1** [comment]
Ill be buying  an A51 soon too. I recently got my iphone stolen so.

**Comment by Janer1** [comment]
Ill try contributing :)

**Comment by TheK0tYaRa** [comment]
So i've got TWRP to build successfully. Now it's the question of fixing `AVB Fail` after flashing recovery.

**Comment by map220v** [comment]
@TheK0tYaRa `avbtool.py add_hash_footer --image image-new.img --partition_name recovery --partition_size <original recovery.img size in bytes>`
This how i signed TWRP for A51 (works with original vbmeta)

**Comment by TheK0tYaRa** [comment]
> @TheK0tYaRa `avbtool.py add_hash_footer --image image-new.img --partition_name recovery --partition_size <original recovery.img size in bytes>`
> This how i signed TWRP for A51 (works with original vbmeta)

It does work for it not to error out. But it doesn't load with only recovery being flashed and is instead stock recovery. There is a compiled boot.img too so i'll flash it with recovery

**Comment by TheK0tYaRa** [comment]
Argh, it worked only the first time! How? Why? Who does that junk? Flashing signed TWRP worked only ONCE and it wasn't bootable. But why tho.....

**Comment by map220v** [comment]
@TheK0tYaRa dtb or recovery_dtbo missing in recovery.img?

**Comment by TheK0tYaRa** [comment]
> @TheK0tYaRa dtb or recovery_dtbo missing in recovery.img?

It was compiled with dtbo.img in the `prebuilt` dir

**Comment by map220v** [comment]
@TheK0tYaRa on Exynos devices it's should have 2 dtbo (dtb and recovery_dtbo) 
https://github.com/map220v/android_device_samsung_a51nsxx/tree/android-10.0/prebuilt

**Comment by TheK0tYaRa** [comment]
@map220v It might not be so. unpack_bootimg returned `recovery dtbo size: 0`

**Comment by TheK0tYaRa** [comment]
It seems that the build process has issues processing dtb and recovery_dtbo and that issue shows in the terminal as missing recovery.fstab . Interesting.

**Comment by TheK0tYaRa** [comment]
Apparently TWRP doesn't yet suppport devices with `super.img` in the AP.
Onward to LineageOS recovery!

**Comment by ShadiestGoat** [comment]
Heya I know this is for a51, but I can confirm this is still a bug on the galaxy A40. Am following this guide: https://topjohnwu.github.io/Magisk/install.html#samsung-system-as-root, and the same issue as the original issue is happening ://

**Comment by soum91** [comment]
@ShadiestGoat I know how frustrated you must be feeling but unfortunately posting here won't do anything. Although it's a opened issue, it's kinda dead. I don't think anyone is working. John or the other contributors are leaving this open because it's not fixed but the issue is kinda abandoned.

**Comment by ShadiestGoat** [comment]
Ah ok, yeah. Its fine though, I used another method which worked in the
end. I did wanna mention that it is still an issue regardless ^^

**Comment by mu198** [comment]
What method did you use? @ShadiestGoat

**Comment by ShadiestGoat** [comment]
I used a boot image instead. So extracted the ap file using 7zip ZS, then
extracted boot.img.extension-i-dont-rmbr, patched that, then compressed the
boot.img into a ap.tar file.
I then odined this tar as AP onto my phone

**Comment by mu198** [comment]
Ok... The method I also use to root the newer devices, since magisk can not patch the new vbmeta(Samsung) with... 
But since it works, the shalt is a new method 😉

**Comment by ShadiestGoat** [comment]
Yis

**Comment by soum91** [comment]
> I used a boot image instead. So extracted the ap file using 7zip ZS, then extracted boot.img.extension-i-dont-rmbr, patched that, then compressed the boot.img into a ap.tar file. I then odined this tar as AP onto my phone
Did that work for you, @ShadiestGoat? Didn't it give **Secure Check Fail: boot.img**?
It gave me that error. I don't know why nothing works with the cursed A20s!

**Comment by ShadiestGoat** [comment]
heya I followed a guide from here, so like you should try following this
instead (im not exactly the best at explaining stuff at 3am
https://forum.xda-developers.com/showpost.php?p=79902328&postcount=29

**Comment by mu198** [comment]
@soum91 I also had the error yesterday on my tablet with the processor described here.

There I had packed the patched boot.img with the 7zZs, 

But when I packed them with the normal 7z it ran, I think there will be a packing error at 7z Zs

**Comment by mu198** [comment]
@ShadiestGoat Yes this is the method by which I can root even the latest KNOX 3.5 devices

**Comment by ShadiestGoat** [comment]
Idk tbh, it still trips up knox but ig thats un avoidant

**Comment by mu198** [comment]
Sry what is idk tbh 😅🤔

**Comment by mu198** [comment]
@ShadiestGoat

**Comment by ShadiestGoat** [comment]
I dont know to be honest....

**Comment by topjohnwu** [comment]
After 2f232fc670018d7061ea4ddd30f8657963c5a8c2 vbmeta is no longer patched

**Comment by soum91** [comment]
@topjohnwu, it still gives **SECURE CHECK FAIL: boot.img** and **SECURE CHECK FAIL: recovery.img**. I used the latest Canary.

**Comment by mu198** [comment]
@soum91 exynos or snapdragon?

**Comment by soum91** [comment]
@mu198, Snapdragon.

**Comment by mu198** [comment]
Ok... Because every new Samsung, with exynos goes the boat.img Method.
Is the snapdragon actually going to unlock the bootloader?

**Comment by soum91** [comment]
Bootloader is already unlocked. John Wu has already flashed magisk to his snapdragon Z Fold.

**Comment by mu198** [comment]
Okay, I think I'm going to ask the same questions as I've ever asked, 😅

1. Do you have the boot.img.lz4 file to boot.img unpacked?
2. You have renamed the magisk_patched.img to boot.img? 
3. You used the normal 7zip to package into a .tar file? Not the 7zZs? 
4. You used the normal Odin 3.14?

**Comment by mu198** [comment]
5. You got the car reboot out at Odin so you can boot straight into the recovery to perform a factory reset? 

@soum91

**Comment by piyush1097** [comment]
Same problem present on Note10 lite (SM-N770F) while flashing magisk patched AP. But patched boot.img do not give any error. Cause of error in flashing Patched AP we are unable to flash twrp or any custom recovery. 
This need to be solved as soon as possible.
I tried disabled vbmeta still it gives error while flashing magisk patched AP and twrp.

**Comment by TheK0tYaRa** [comment]
Got a brand new error after flashing! https://del.dog/phivinuppe
Flashed by the new instruction. I remember that it was needed to flash HOME_CSC before.
For short - it's HASH_MISMATCH

**Comment by soum91** [comment]
@TheK0tYaRa HOME_CSC used to be needed but John now have instructed to use CSC as using it causes data wipe. But the problem is lower end device(at least SM-A207F) don't have CSC file.

As I mentioned in issue #3231 patched file flashes fine but causes **Secure Check Fail: boot.img** screen bootloops. With 8.0.1 magisk is supposed to be fixing some kind of bootloops but not sure if that is this one. I asked John on Twitter to confirm it but he didn't reply me for once. And I don't want take the pain in the ass of setting up the phone for the zillionth time if it's still not fixed.

**Comment by TheK0tYaRa** [comment]
I have used a file that has no HOME_ in it's name to be clear. ROM has both.

**Comment by piyush1097** [comment]
Im also getting HASH_MISMATCH error after flashing patched AP by magisk v8.0.1 on Note10 lite (SM-N770F)

**Comment by coderpradp** [comment]
@TheK0tYaRa do you have any update? I'm getting the vbmeta mismatch too on Samsung M21F on Magisk Manager 8.0.7
Afaik, newer magisk manager should not be patching vbmeta, but on my log, it says patching vbmeta.img

**Comment by jjgh** [comment]
Having the same issue @TheK0tYaRa mentioned here on a Samsung a40 after flashing new Magisk-patched images according to instructions https://twrp.me/samsung/samsunggalaxya40.html
Does anyone have a solution? 
(can't believe this is in this state after more than one year?)

**Comment by chipAngel5G** [comment]
> Hi
> When using AP magisk patched tar it makes vbmeta error with dowload screen:
> ODIN MODE (AVB Fail), vbmeta: error verifying vbmeta image: OK_NOT_SIGNED, VERIFICATION_DISABLED bit is set, custom vbmeta No sign info, vbmeta .
> [magisk_install_log_KEEPON.log](https://github.com/topjohnwu/Magisk/files/4071184/magisk_install_log_KEEPON.log)
> [magisk_install_log_KEEPOFF.log](https://github.com/topjohnwu/Magisk/files/4071185/magisk_install_log_KEEPOFF.log)
> 
> SM-A515F with A515FXXU1ASKJ firmware Android 10 has 2 vbmeta: vbmeta.img and vbmeta_samsung.img in AP.
> [vbmeta_stock_imgs.zip](https://github.com/topjohnwu/Magisk/files/4071199/vbmeta_stock_imgs.zip)
> 
> Only flashing boot.img makes bootloop.
> Magisk versions tested from 19.3, 20.3 to latest canary build.
> 
> TWRP is not available. The error appears after flashing and changing vbmeta.img. Bootloader is unlocked, KG is checking. With original vbmeta magisk recovery is not accepted.
> 
> What should be done to work?

Off the subject..but how did you unlock your bootloader?
